src/misceval.js

```diff
@@ -1189,6 +1189,14 @@ Sk.misceval.applyOrSuspend = function (func, kwdict, varargseq, kws, args) {
         //append kw args to args, filling in the default value where none is provided.
         return func.apply(null, args);
     } else {
+        fcall = func.__get__;
+        if (fcall !== undefined) {
+            fcall = Sk.misceval.callsimOrSuspend(func.__get__, func, func);
+            //args.unshift(func);
+            if (fcall.tp$call) {
```

Comment:
So, why is `func` a suspension? I'm guessing that it's because it's decorated, and the decorator suspended?

If this happens, we shouldn't continue and make the value of the function variable a suspension. What should happen is that the decorator should be allowed to complete before that variable gets assigned. Can you show me where the `compile.js` output for decorators is?